src/main/java/com/denizenscript/denizencore/scripts/commands/core/RedisCommand.java

```diff
@@ -0,0 +1,449 @@
+                    return;
+                }
+                if (subscriptions.containsKey(redisID)) {
+                    Debug.echoError(scriptEntry.getResidingQueue(), "Cannot publish messages while subscribed to redis server with ID '" + redisID + "'!");
+                    scriptEntry.setFinished(true);
+                    return;
+                }
+                Debug.echoDebug(scriptEntry, "Publishing message '" + message.asString() + "' to channel '" + channel.asString() + "'");
+                Runnable doQuery = () -> {
+                    try {
+                        ElementTag result = new ElementTag(con.publish(CoreUtilities.toLowerCase(channel.asString()), message.asString()));
+                        scriptEntry.addObject("result", result);
+                        DenizenCore.schedule(new OneTimeSchedulable(() -> {
+                            scriptEntry.setFinished(true);
+                        }, 0));
+                    }
+                    catch (final Exception ex) {
+                        DenizenCore.schedule(new OneTimeSchedulable(() -> {
+                            Debug.echoError(scriptEntry.getResidingQueue(), "Redis Exception: " + ex.getMessage());
+                            scriptEntry.setFinished(true);
+                            if (Debug.verbose) {
+                                Debug.echoError(scriptEntry.getResidingQueue(), ex);
+                            }
+                        }, 0));
+                    }
+                };
+                if (scriptEntry.shouldWaitFor()) {
+                    DenizenCore.schedule(new AsyncSchedulable(new OneTimeSchedulable(doQuery, 0)));
+                }
+                else {
+                    doQuery.run();
+                }
+            }
+            else if (action.asString().equalsIgnoreCase("command")) {
+                if (command == null) {
+                    Debug.echoError(scriptEntry.getResidingQueue(), "Must specify a valid redis command!");
+                    scriptEntry.setFinished(true);
+                    return;
+                }
+                final Jedis con = connections.get(redisID);
+                if (con == null) {
+                    Debug.echoError(scriptEntry.getResidingQueue(), "Not connected to redis server with ID '" + redisID + "'!");
+                    scriptEntry.setFinished(true);
+                    return;
+                }
+                if (subscriptions.containsKey(redisID)) {
+                    Debug.echoError(scriptEntry.getResidingQueue(), "Cannot run commands while subscribed to redis server with ID '" + redisID + "'!");
+                    scriptEntry.setFinished(true);
+                    return;
+                }
+                Debug.echoDebug(scriptEntry, "Running command " + command.asString());
+                Runnable doQuery = () -> {
+                    try {
+                        String[] args = ArgumentHelper.buildArgs(command.asString());

```

Comment:
This seems very likely to have side effects - consider for example:
`- redis id:name "command:setex my_key 60 'message <context.message>'"`
If the message is `bobby' drop table students '` or somethin, yknow. If there's not a better way of doing this, it should at least be documented that any raw input values should usually be escaped.